src/Arcus.WebApi.Unit/Security/Authentication/SharedAccessKeyAuthenticationAttributeTests.cs

```diff
@@ -0,0 +1,150 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Arcus.Security.Secrets.Core.Interfaces;
+using Arcus.WebApi.Security.Authentication;
+using Arcus.WebApi.Unit.Hosting;
+using Microsoft.AspNetCore.Mvc;
+using Xunit;
+
+namespace Arcus.WebApi.Unit.Security.Authentication
+{
+    public class SharedAccessKeyAuthenticationAttributeTests : IDisposable
+    {
+        private const string HeaderName = "x-shared-access-key",
+                             SecretName = "custom-access-key-name",
+                             AuthorizedRoute = "/authz/shared-access-key";
+
+        private readonly TestApiServer _testServer = new TestApiServer();
+
+        [Theory]
+        [InlineData(null, "not empty or whitespace")]
+        [InlineData("", "not empty or whitespace")]
+        [InlineData(" ", "not empty or whitespace")]
+        [InlineData("not empty or whitespace", null)]
+        [InlineData("not empty or whitespace", "")]
+        [InlineData("not empty or whitespace", " ")]
+        public void SharedAccessKeyAttributeAndFilter_WithNotPresentHeaderNameAndOrSecretName_ShouldFailWithArgumentException(
+            string headerName,
+            string secretName)
+        {
```

Comment:
Shouldn't this be splitted into two tests actually?

Reply:
Could be, but as testing a single 'unit' I guess it would be logical to test the inputs together on default argument exceptions.

Reply:
I'm not sure if I agree as you are testing validation for 2 classes in one test?

Reply:
Ah, ow; didn't see that. Yes. I'll split that.

Reply:
No problem, that's why we have reviews!